src/mpi/romio/test/file_info.c

```diff
@@ -412,6 +411,12 @@ int main(int argc, char **argv)
     }
 
     MPI_File_close(&fh);
+
+    /* delete the file */
+    if (!mynod)
+        MPI_File_delete(filename, MPI_INFO_NULL);
```

Comment:
as you observe, if N processes try to call MPI_File_delete, one will succeed and N-1 will fail.  Error handling in ROMIO tests is fairly primitive but here we deliberatey did not check for erorr.  Does your driver give a fatal error if deleting a thing that does not exist?

Reply:
i think i was checking something here and this is not required actually.
no the driver doesn't do anything fatal.

Reply:
fwiw i added this delete. the test did not actually delete the file..
other tests that delete file do issue the delete from rank 0 only.